internal/signing/channelaccounts/channel_accounts_model_test.go

```diff
@@ -0,0 +1,211 @@
+package channelaccounts
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stellar/go/keypair"
+	"github.com/stellar/wallet-backend/internal/db"
+	"github.com/stellar/wallet-backend/internal/db/dbtest"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func createChannelAccountFixture(t *testing.T, ctx context.Context, dbConnectionPool db.ConnectionPool, publicKey, encryptedPrivateKey string) {
+	t.Helper()
+	const q = `
+		INSERT INTO 
+			channel_accounts (public_key, encrypted_private_key)
+		VALUES
+			($1, $2)
+	`
+	_, err := dbConnectionPool.ExecContext(ctx, q, publicKey, encryptedPrivateKey)
+	require.NoError(t, err)
+}
+
+func TestChannelAccountModelGetIdleChannelAccount(t *testing.T) {
+	dbt := dbtest.Open(t)
+	defer dbt.Close()
+
+	dbConnectionPool, err := db.OpenDBConnectionPool(dbt.DSN)
+	require.NoError(t, err)
+	defer dbConnectionPool.Close()
+
+	ctx := context.Background()
+	m := NewChannelAccountModel(dbConnectionPool)
+
+	t.Run("returns_error_when_there's_no_channel_account_available", func(t *testing.T) {
+		channelAccount1 := keypair.MustRandom()
+		channelAccount2 := keypair.MustRandom()
+		createChannelAccountFixture(t, ctx, dbConnectionPool, channelAccount1.Address(), channelAccount1.Seed())
+		createChannelAccountFixture(t, ctx, dbConnectionPool, channelAccount2.Address(), channelAccount2.Seed())
+
+		const lockChannelAccountQuery = `
+			UPDATE
+				channel_accounts
+			SET
+				locked_at = NOW(),
+				locked_until = NOW() + '5 minutes'::INTERVAL
+			WHERE
+				public_key = $1
+		`
+		_, err := dbConnectionPool.ExecContext(ctx, lockChannelAccountQuery, channelAccount1.Address())
+		require.NoError(t, err)
+		_, err = dbConnectionPool.ExecContext(ctx, lockChannelAccountQuery, channelAccount2.Address())
+		require.NoError(t, err)
+
+		ca, err := m.GetIdleChannelAccount(ctx, time.Minute)
+		assert.ErrorIs(t, err, ErrNoIdleChannelAccountAvailable)
+		assert.Nil(t, ca)
+	})
+
+	t.Run("returns_error_when_there's_no_channel_account_available", func(t *testing.T) {
```

Comment:
```suggestion
	t.Run("returns_channel_account_available", func(t *testing.T) {
```
Interesting that Go does not complain about subtests with duplicate names.